hscommon/hsenum/weapon_class_string.go

```diff
@@ -0,0 +1,46 @@
+package hsenum
+
+import (
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+)
+
+// GetWeaponClassString returns full name of weapon class given
+// nolint:gocyclo // can't reduce
+func GetWeaponClassString(cls d2enum.WeaponClass) string {
+	var weapon string
+
+	switch cls {
+	case d2enum.WeaponClassNone:
+		weapon = stringNone
+	case d2enum.WeaponClassHandToHand:
+		weapon = "Hand To Hand"
+	case d2enum.WeaponClassBow:
+		weapon = "Bow"
+	case d2enum.WeaponClassOneHandSwing:
+		weapon = "One Hand Swing"
+	case d2enum.WeaponClassOneHandThrust:
+		weapon = "One Hand Thrust"
+	case d2enum.WeaponClassStaff:
+		weapon = "Staff"
+	case d2enum.WeaponClassTwoHandSwing:
+		weapon = "Two Hand Swing"
+	case d2enum.WeaponClassTwoHandThrust:
+		weapon = "Two Hand Thrust"
+	case d2enum.WeaponClassCrossbow:
+		weapon = "Crossbow"
+	case d2enum.WeaponClassLeftJabRightSwing:
+		weapon = "Left Jab Right Swing"
+	case d2enum.WeaponClassLeftJabRightThrust:
+		weapon = "Left Jab Right Thrust"
+	case d2enum.WeaponClassLeftSwingRightSwing:
+		weapon = "Left Swing Right Swing"
+	case d2enum.WeaponClassLeftSwingRightThrust:
+		weapon = "Left Swing Right Thrust"
+	case d2enum.WeaponClassOneHandToHand:
+		weapon = "One Hand To Hand"
+	case d2enum.WeaponClassTwoHandToHand:
+		weapon = "Two Hand To Hand"
+	}
+
+	return weapon
```

Comment:
Same here; we can avoid the need for a nolint directive if we use a map:
```golang
func GetWeaponClassString(weaponClass d2enum.WeaponClass) string {
	const (
		strNone    = "None"
		strUnknown = "Unknown"
	)

	weaponClassNames := map[d2enum.WeaponClass]string{
		d2enum.WeaponClassNone:                 strNone,
		d2enum.WeaponClassHandToHand:           "Hand To Hand",
		d2enum.WeaponClassBow:                  "Bow",
		d2enum.WeaponClassOneHandSwing:         "One Hand Swing",
		d2enum.WeaponClassOneHandThrust:        "One Hand Thrust",
		d2enum.WeaponClassStaff:                "Staff",
		d2enum.WeaponClassTwoHandSwing:         "Two Hand Swing",
		d2enum.WeaponClassTwoHandThrust:        "Two Hand Thrust",
		d2enum.WeaponClassCrossbow:             "Crossbow",
		d2enum.WeaponClassLeftJabRightSwing:    "Left Jab Right Swing",
		d2enum.WeaponClassLeftJabRightThrust:   "Left Jab Right Thrust",
		d2enum.WeaponClassLeftSwingRightSwing:  "Left Swing Right Swing",
		d2enum.WeaponClassLeftSwingRightThrust: "Left Swing Right Thrust",
		d2enum.WeaponClassOneHandToHand:        "One Hand To Hand",
		d2enum.WeaponClassTwoHandToHand:        "Two Hand To Hand",
	}

	weaponClassName, found := weaponClassNames[weaponClass]
	if !found {
		return strUnknown
	}

	return weaponClassName
}
```

Reply:
honestly, I think it might be worthwhile to add methods to our enum types to avoid having to do this:
```golang
func (wc WeaponClass) DescriptiveName() string {
	const (
		strNone    = "None"
		strUnknown = "Unknown"
	)
	
	descriptions := map[WeaponClass]string{
		WeaponClassNone:                 strNone,
		WeaponClassHandToHand:           "Hand To Hand",
		WeaponClassBow:                  "Bow",
		WeaponClassOneHandSwing:         "One Hand Swing",
		WeaponClassOneHandThrust:        "One Hand Thrust",
		WeaponClassStaff:                "Staff",
		WeaponClassTwoHandSwing:         "Two Hand Swing",
		WeaponClassTwoHandThrust:        "Two Hand Thrust",
		WeaponClassCrossbow:             "Crossbow",
		WeaponClassLeftJabRightSwing:    "Left Jab Right Swing",
		WeaponClassLeftJabRightThrust:   "Left Jab Right Thrust",
		WeaponClassLeftSwingRightSwing:  "Left Swing Right Swing",
		WeaponClassLeftSwingRightThrust: "Left Swing Right Thrust",
		WeaponClassOneHandToHand:        "One Hand To Hand",
		WeaponClassTwoHandToHand:        "Two Hand To Hand",
	}

	description, found := descriptions[wc]
	if !found {
		return strUnknown
	}
	
	return description
}
```

However, this would need to be done in the `OpenDiablo2` repo